test/unit/discovery_rule_test.rb

```diff
@@ -172,6 +172,27 @@ class DiscoveryRuleTest < ActiveSupport::TestCase
     assert_equal "Host group location #{loc.name} must also be associated to the discovery rule", rule.errors[:locations].first
   end
 
+  test 'should be able to suggest next priority with orgs' do
+      # create two different sets of taxamonies
+      organization1 = FactoryBot.create(:organization, :name => 'Organization one')
+      FactoryBot.create(:location, :name => 'Location one')
+
+      organization2 = FactoryBot.create(:organization, :name => 'Organization two')
+      FactoryBot.create(:location, :name => 'Location two')
+
+      # create two differenct hostgroups
+      hostgroup1 = FactoryBot.create(:hostgroup, organizations: Organization.where(name: 'Organization one'), locations: Location.where(name: 'Location one'))
+      hostgroup2 = FactoryBot.create(:hostgroup, organizations: Organization.where(name: 'Organization two'), locations: Location.where(name: 'Location two'))
+
+      # create discovery_rules using difffent taxonomies
+      FactoryBot.create(:discovery_rule, name: "NewRule1", :priority => 100, hostgroup: hostgroup1, organizations: Organization.where(name: 'Organization one'), locations: Location.where(name: 'Location one'))
+
+      FactoryBot.create(:another_discovery_rule, name: "NewRule", :priority => 100, hostgroup: hostgroup2, organizations: Organization.where(name: 'Organization two'), locations: Location.where(name: 'Location two'))
+
+      # check if both have equal suggested priorities.
+      assert_equal DiscoveryRule.suggest_priority(organization2), DiscoveryRule.suggest_priority(organization1)
+  end
```

Comment:
Also check the actual value to verify it's not like `-1` or something like that. Also adding one more test where you try to increase priority for org A but suggested priority is unaffected for org B would be useful. Please use test contexts to wrap setup code.